Отрисуй меня полностью (часть 1)

lisenkovnikita opened the pull request:
modified:   js/data.js
modified:   js/main.js
new file:   js/rendering.js
modified:   js/util.js

---
:mortar_board: [Отрисуй меня полностью (часть 1)](https://up.htmlacademy.ru/javascript/27/user/2054309/tasks/12)

:boom: https://htmlacademy-javascript.github.io/2054309-kekstagram-simple-27/5/

lisenkovnikita committed:
Отрисуй меня полностью (часть 1)
modified:   js/data.js
modified:   js/main.js
new file:   js/rendering.js
modified:   js/util.js

frontstall commented on js/data.js:
```diff
@@ -18,16 +15,16 @@ const DESCRIPTION = [
 ];
 
-const generateRandomPhoto = () => {
+function generateRandomPhoto() {
```
стрелочная функция

frontstall commented on js/data.js:
```diff
@@ -18,16 +15,16 @@ const DESCRIPTION = [
-const generateRandomPhoto = () => {
+function generateRandomPhoto() {
   const photo = {};
   photo.id = getId();
```
мы же обсуждали, по условию айдишник от 1 до 25, и даже реализовали функции getPictureData и getPicturesData, которые используют в качестве айдишников счётчик в цикле.

frontstall commented on js/data.js:
```diff
@@ -18,16 +15,16 @@ const DESCRIPTION = [
-};
+}
 
 const generateGroupOfPhotos = () => Array.from({length: PHOTOS_AMOUNT}, generateRandomPhoto);
```
эта функция не генерирует группу фотографий, она генерирует данные для фотографий. Помнишь, мы обсуждали то, как называть функции?

frontstall commented on js/main.js:
```diff
@@ -1,3 +1,2 @@
-
-console.log(generateGroupOfPhotos());
+import {startRenderind} from './rendering.js';
+startRenderind();
```
`startRenderind` не подходящее название. Почему "начать рендеринг"? Т.е. будет и продолжение рендеринга? Рендеринга чего? В задании нужно отрендерить фотографии.

frontstall commented on js/rendering.js:
```diff
@@ -0,0 +1,23 @@
+import {generateGroupOfPhotos} from './data.js';
+
+function startRenderind () {
```
стрелочная функция

frontstall commented on js/rendering.js:
```diff
@@ -0,0 +1,23 @@
+
+function startRenderind () {
+  const pictures = document.querySelector('.pictures');
+  const templePicture = document.querySelector('#picture')
```
template

temple – это "храм" по-английски :)

frontstall commented on js/rendering.js:
```diff
@@ -0,0 +1,23 @@
+
+  groupPhotos.forEach(({ url, likes, commentsAmount }) => {
+    const randomElement = templePicture.cloneNode(true);
+    randomElement.querySelector('.picture__img').src = url;
+    randomElement.querySelector('.picture__likes').textContent = likes;
+    randomElement.querySelector('.picture__comments').textContent = commentsAmount;
```
как мы и говорили, создание элементов нужно отделить от собственно вывода на экран

frontstall commented on js/util.js:
```diff
@@ -24,4 +24,4 @@ const getId = () => {
 };
 
-export {getRandomArrayElement, getUrl, getId, getRandomPositiveInteger};
+export {getRandomArrayElement, getUrl, getId, getRandomPositiveInteger, id};
```
зачем экспортировать id? в этой переменной примитив, передаётся по значению.

lisenkovnikita committed:
Отрисуй меня полностью (часть 1)
modified:   js/data.js
modified:   js/main.js
modified:   js/rendering.js
modified:   js/util.js

lisenkovnikita committed:
Отрисуй меня полностью (часть 1)
modified:   js/data.js
modified:   js/rendering.js

lisenkovnikita committed:
Отрисуй меня полностью (часть 1)
modified:   js/data.js
modified:   js/main.js
modified:   js/rendering.js
modified:   js/util.js
